04/GeoffRiley/usertweets.py

```diff
@@ -0,0 +1,68 @@
+from datetime import datetime
+from collections import namedtuple
+import csv
+import os
+
+import tweepy
+
+from config import CONSUMER_KEY, CONSUMER_SECRET
+from config import ACCESS_TOKEN, ACCESS_SECRET
+
+DEST_DIR = 'data'
+EXT = 'csv'
+NUM_TWEETS = 100
+
+Tweet = namedtuple('Tweet', 'id_str created_at text')
+
+
+class UserTweets(object):
+    """TODOs:
+    - create a tweepy api interface
+    - get all tweets for passed in handle
+    - optionally get up until 'max_id' tweet id
+    - save tweets to csv file in data/ subdirectory
+    - implement len() an getitem() magic (dunder) methods"""
+
+    def __init__(self, handle, max_id=None):
+        self._handle = handle
+        self._save_file = f'{DEST_DIR}/{handle}.{EXT}'
+        self._max_id = max_id
+        self._auth = tweepy.OAuthHandler(CONSUMER_KEY, CONSUMER_SECRET)
+        self._auth.set_access_token(ACCESS_TOKEN, ACCESS_SECRET)
+        self._api = tweepy.API(self._auth)
```

Comment:
could make a helper

Reply:
Yes, I understand that concept. My reason for not doing so was that as it's a self contained object it wasn't going to be changing any of those elements after the initialisation.

However, one development I did consider was to make the authorisation and connection as a separate singleton class in order to isolate that part of the activity… but it seemed a little overkill for this example :)